server/src/test/java/io/pravega/schemaregistry/storage/impl/group/GroupTest.java

```diff
@@ -0,0 +1,592 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.schemaregistry.storage.impl.group;
+
+import com.google.common.collect.ImmutableMap;
+import io.pravega.schemaregistry.common.Either;
+import io.pravega.schemaregistry.contract.data.Compatibility;
+import io.pravega.schemaregistry.contract.data.EncodingId;
+import io.pravega.schemaregistry.contract.data.EncodingInfo;
+import io.pravega.schemaregistry.contract.data.GroupHistoryRecord;
+import io.pravega.schemaregistry.contract.data.GroupProperties;
+import io.pravega.schemaregistry.contract.data.SchemaInfo;
+import io.pravega.schemaregistry.contract.data.SchemaWithVersion;
+import io.pravega.schemaregistry.contract.data.SerializationFormat;
+import io.pravega.schemaregistry.contract.data.VersionInfo;
+import io.pravega.schemaregistry.storage.Etag;
+import io.pravega.schemaregistry.storage.StoreExceptions;
+import io.pravega.schemaregistry.storage.impl.group.records.TableRecords;
+import io.pravega.test.common.AssertExtensions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class GroupTest {
+    String groupId;
+    InMemoryGroupTable integerGroupTable;
+    Group<Integer> integerGroup;
+    private ScheduledExecutorService executor;
+
+    @Before
+    public void setUp() {
+        executor = Executors.newScheduledThreadPool(5);
+        groupId = "mygroup";
+        integerGroupTable = new InMemoryGroupTable();
+        integerGroup = new Group<Integer>(integerGroupTable, executor);
+    }
+
+    @After
+    public void tearDown() {
+        executor.shutdownNow();
+    }
+
+    @Test
+    public void testCreate() {
+        integerGroup.create(SerializationFormat.Custom, ImmutableMap.of(), Boolean.TRUE,
```

Comment:
here and everywhere:
add cases to cover failing cases -- 
1. simulate 2 concurrent requests, both being sent with same etag..
first one should complete while the other should fail with WriteConflict.